clicache/src/DataInput.hpp

```diff
@@ -271,10 +286,15 @@ namespace Apache
         List<Object^>^ ReadObjectArray();
 
         /// <summary>
-        /// Read a array of signed byte array from the stream.
+        /// Read an array of unsigned byte arrays from the stream.
```

Comment:
Are you really thinking about changing the behavior of the public API method to return unsigned bytes?

Reply:
No. The intent is merely to add support for currently unsupported types in .NET. Will make sure that I don't change any existing APIs. Just waiting on my superfly to complete. Hopefully will close this PR and open a new one today.

Reply:
This was a good catch. I did have some breaking changes in the API, which have now been fixed. All unsigned types are through new APIs.